repository/util.go

```diff
@@ -0,0 +1,128 @@
+package repository
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+var (
+	urlSchemeRegExp  = regexp.MustCompile(`^[^:]+://`)
+	scpLikeURLRegExp = regexp.MustCompile(`^(?:(?P<user>[^@]+)@)?(?P<host>[^:\s]+):(?:(?P<port>[0-9]{1,5})(?:\/|:))?(?P<path>[^\\].*\/[^\\].*)$`)
+)
+
+func getContainerUseRemote(ctx context.Context, repo string) (string, error) {
+	// Check if we already have a container-use remote
+	cuRemote, err := runGitCommand(ctx, repo, "remote", "get-url", "container-use")
+	if err != nil {
+		if strings.Contains(err.Error(), "No such remote") {
+			return "", os.ErrNotExist
+		}
+		return "", err
+	}
+
+	return strings.TrimSpace(cuRemote), nil
+}
+
+func normalizeForkPath(ctx context.Context, repo string) (string, error) {
+	// Check if there's an origin remote
+	origin, err := runGitCommand(ctx, repo, "remote", "get-url", "origin")
+	if err != nil {
+		// If not -- this repository is a local one, we're going to use the filesystem path for the container-use repo
+		if strings.Contains(err.Error(), "No such remote") {
+			return homedir.Expand(filepath.Join(cuRepoPath, repo))
+		}
+		return "", err
+	}
+
+	// Otherwise, let's use the normalized origin as path
+	normalizedOrigin, err := normalizeGitURL(strings.TrimSpace(origin))
+	if err != nil {
+		return "", err
+	}
+	return homedir.Expand(filepath.Join(cuRepoPath, normalizedOrigin))
+}
+
+func normalizeGitURL(endpoint string) (string, error) {
```

Comment:
This boilerplate was copied from dagger, which in turn was copied from Go.

It simply normalizes e.g. `https://github.com/dagger/container-use.git` and `git@github.com:dagger/container-use.git` both to `github.com/dagger/container-use`